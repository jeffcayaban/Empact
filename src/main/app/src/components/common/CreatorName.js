import React from "react";
import { Link } from "react-router-dom";

/**
 * Displays the creator name and a link to their profile.
 */

class CreatorName extends React.PureComponent {

    render() {
        const content = this.props.content;

        if (!content.isAnonymous) {
            // Creator chose to show name.
            const createdBy = content.createdBy;
            return (
                <span><Link to={`/user/${createdBy.username}`}>{`${createdBy.firstName} ${createdBy.lastName}`}</Link></span>
            )

        } else {
            // Creator decides to stay anonymous.
            return <span>Anonymous</span>
        }
    }
}

export default CreatorName;