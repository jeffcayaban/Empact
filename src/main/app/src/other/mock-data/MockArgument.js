import {CRITICAL_QUESTION_IDS, EXPERT_OPINION, ROLE_ADMIN} from "../../constants";

export default {
    id: 'testId',
    isSupporting: true,
    rootPetitionId: 'testRootPetitionId',
    parentArgumentId: 'testParentArgumentId',
    subCriticalQuestionId: null,
    agreesWithCQ: true,
    criticalQuestionId: CRITICAL_QUESTION_IDS[0],
    creationDateTime: 1551716401541,
    lastUpdatedDateTime: 1551716402000,
    createdBy: {
        id: 'testId',
        username: 'testUsername',
        firstName: 'testFirstName',
        lastName: 'testLastName',
        authorities: [ROLE_ADMIN]
    },
    argumentType: EXPERT_OPINION,
    isAnonymous: false,
    sources: [],
    expert: "testExpert",
    expertDomain: 'testDomain',
    expertAssertion: 'testAssertion'
};