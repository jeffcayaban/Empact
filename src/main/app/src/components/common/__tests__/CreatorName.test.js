import React from 'react';
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";

import MockArgument from '../../../other/mock-data/MockArgument';
import CreatorName from "../CreatorName";

describe('CreatorName', () => {

    it('should render successfully', () => {
        const component = shallow(<CreatorName content={MockArgument} />);
        expect(toJson(component)).toMatchSnapshot();
    });

    it('should render successfully with Anonymous', () => {
        MockArgument.isAnonymous = true;

        const component = shallow(<CreatorName content={MockArgument} />);
        expect(toJson(component)).toMatchSnapshot();
    });

});
