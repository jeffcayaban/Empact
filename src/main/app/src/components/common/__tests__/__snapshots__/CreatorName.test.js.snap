// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatorName should render successfully 1`] = `
<span>
  <Link
    replace={false}
    to="/user/testUsername"
  >
    testFirstName testLastName
  </Link>
</span>
`;

exports[`CreatorName should render successfully with Anonymous 1`] = `
<span>
  Anonymous
</span>
`;
